src/main.c

```diff
@@ -137,7 +137,9 @@ static gboolean check_graphical_sessions(gpointer data) {
 
 static void do_startup(GObject *object, gpointer data) {
   GError *error = NULL;
-  notify_init("snapd-desktop-integration");
+#ifndef USE_GNOTIFY
```

Comment:
Where does this define come from? Why do we not always use this?

Reply:
As this patch description says, there is a problem with gnotify and snapd, so it can't identify the right .desktop file and doesn't set the top-left icon in the notifications. Changing to libnotify instead allows to set that icon, which is what this patch does.

I added an screen capture with the problem in the top description.

Reply:
It's probably useful to add a comment here indicating why this is optional / not working.

Reply:
Added.